### PR TITLE
[refactor] Use multiline strings for better readability

### PR DESCRIPTION
**Pull request by ska80**
This PR is a follow-up to #1142 as I couldn't reopen it.

### CODE REVIEW

**Comment by andrevidela** [comment]
May I suggest reverting to `unlines`, merging this and opening an issue ?

**Comment by ska80** [comment]
> May I suggest reverting to `unlines`, merging this and opening an issue ?

Good idea.

File: src/Compiler/RefC/RefC.idr
Comment (mattpolzin) [inline src/Compiler/RefC/RefC.idr:455]: Although I can see the temptation, I am not sure if we should be making arbitrary whitespace changes given that we don't have community or compiler-specific guidelines on the subject and the author might well have made intentional choices around these newlines based on their personal preferences.

File: src/Compiler/Scheme/Chez.idr
Comment (mattpolzin) [inline src/Compiler/Scheme/Chez.idr:410]:
```suggestion
  export LD_LIBRARY_PATH="$DIR/\{ appdir }:$LD_LIBRARY_PATH"
  export DYLD_LIBRARY_PATH="$DIR/\{ appdir }:$DYLD_LIBRARY_PATH"
```

File: src/Compiler/Scheme/Chez.idr
Comment (mattpolzin) [inline src/Compiler/Scheme/Chez.idr:420]: this is new. I know so little about Chez that I can't say whether I like it or not :) but it's not a direct translation.

File: src/Compiler/Scheme/Chez.idr
Comment (mattpolzin) [inline src/Compiler/Scheme/Chez.idr:437]:
```suggestion
  PATH="$DIR/\{ appdir }:$PATH"
```

File: src/Idris/Package.idr
Comment (mattpolzin) [inline src/Idris/Package.idr:798]: Left in by accident?

File: src/Idris/CommandLine.idr
Comment (mattpolzin) [inline src/Idris/CommandLine.idr:411]: This creates an extra newline right?

File: src/Core/Options.idr
Comment (mattpolzin) [inline src/Core/Options.idr:44]: I suppose I don't personally mind one way or the other, but why the change to not quoting these strings (wdir, bdir, ldir, dfix)?

**Review by mattpolzin — COMMENTED** [review]
Looked over most changes for consistency. Didn't quite make it through all of the Chez compiler files yet because with no Chez background it takes me some time to parse whether any unintended changes have been introduced.

File: src/Compiler/Scheme/Chez.idr
Comment (ska80) [reply src/Compiler/Scheme/Chez.idr:420]: This change just adds a comment to a batch script with CG's name.

File: src/Compiler/Scheme/Chez.idr
Comment (ska80) [reply src/Compiler/Scheme/Chez.idr:437]: Shouldn't we just assume that the rest of the `$PATH` should be already correctly escaped, and we only care about the path that we prepend?

File: src/Idris/Package.idr
Comment (ska80) [reply src/Idris/Package.idr:798]: See issue #2087 

I decided to keep the multiline version.  Once #2087 is fixed, it can then be used in place of the 'unlines' version. If it is not acceptable I will delete it.

File: src/Idris/CommandLine.idr
Comment (ska80) [reply src/Idris/CommandLine.idr:411]: This is a direct translation.  With every change, I tested the output to make sure it was the same.

File: src/Compiler/Scheme/Chez.idr
Comment (ska80) [reply src/Compiler/Scheme/Chez.idr:410]: I tried to keep a direct translation.

File: src/Compiler/RefC/RefC.idr
Comment (ska80) [reply src/Compiler/RefC/RefC.idr:455]: Fair enough.  I can revert those vertical whitespace changes if needed.  I agree I made it look more pleasing according to my personal preferences. We do need community guidelines.

File: src/Idris/CommandLine.idr
Comment (ska80) [reply src/Idris/CommandLine.idr:411]: My bad, you are right.  Will fix it.

File: src/Core/Options.idr
Comment (ska80) [reply src/Core/Options.idr:44]: Will fix.

File: src/Compiler/Scheme/ChezSep.idr
Comment (mattpolzin) [inline src/Compiler/Scheme/ChezSep.idr:48]: Missing `(load-shared-object \"ws2_32.dll\")`

File: src/Idris/Package.idr
Comment (mattpolzin) [reply src/Idris/Package.idr:798]: I think we should remove the comment in the meantime. Until the multiline version is actually made to be the version in-code, it creates an extra place where any modifications to the message needs to be made and for the same reason the commented out version will need to be checked for consistency after the bug is fixed so it doesn't really save time at that point IMO.

File: src/Compiler/RefC/RefC.idr
Comment (mattpolzin) [reply src/Compiler/RefC/RefC.idr:455]: Personally, I would say these newline changes should be reverted before merging, but if you would like to keep them and another maintainer decides this is ready to merge with these changes in place, I won't put up a fuss over such a minor detail. I just consider it a courtesy to the original author of the file.

**Review by mattpolzin — CHANGES_REQUESTED** [review]
I've looked over the remaining files, so to the extent that my eyes can be trusted to spot things, I've only found 1 remaining meaningful difference.

Looks really close!

File: src/Compiler/Scheme/ChezSep.idr
Comment (ska80) [reply src/Compiler/Scheme/ChezSep.idr:48]: I removed this line due to the removal of the same line in `Chez.idr`.

File: src/Compiler/RefC/RefC.idr
Comment (ska80) [reply src/Compiler/RefC/RefC.idr]: Ok, I have reverted these whitespace changes.

File: src/Idris/Package.idr
Comment (ska80) [reply src/Idris/Package.idr]: Done.  Thanks for the review!

**Review by mattpolzin — APPROVED** [review]
This is looking ready to me! Many of those blocks of text are definitely easier to read using multiline string literals.

File: src/Compiler/Scheme/Racket.idr
Comment (andrevidela) [inline src/Compiler/Scheme/Racket.idr:360]: Is this line extra?

**Comment by andrevidela** [comment]
I went through it too, didn't notice anything except for 1 comment

File: src/Compiler/Scheme/Racket.idr
Comment (ska80) [reply src/Compiler/Scheme/Racket.idr:360]: Yup

Comment (andrevidela) [reply src/Compiler/Scheme/Racket.idr:360]: should it be removed? I'm not clear why it's been added

Comment (ska80) [reply src/Compiler/Scheme/Racket.idr:360]: Are we talking about the vertical whitespace or the comment with a function that inserts CG's name?

If it is the latter, all shell startup wrapper scripts have this function that inserts CG's name, so I added it here for consistency.

Comment (andrevidela) [reply src/Compiler/Scheme/Racket.idr:360]: It was the latter, seems good to me!

Comment (ska80) [reply src/Compiler/Scheme/Racket.idr:360]: @andrevidela Thanks for the review!

**Pull request merged**
